src/com/oltpbenchmark/Phase.java

```diff
@@ -126,108 +91,124 @@ public boolean isThroughputRun() {
     }
 
     public void resetSerial() {
-        if (this.serial) {
-            this.nextSerials.clear();
-            this.nextSerials.addAll(this.startSerials);
-        }
+        this.nextSerial = 1;
     }
 
     public int getActiveTerminals() {
         return activeTerminals;
     }
 
     public int getWeightCount() {
-        return (this.num_weights);
+        return (this.weightCount);
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public int getTime() {
+        return time;
     }
+
+    public int getWarmupTime() {
+        return warmupTime;
+    }
+
+    public int getRate() {
+        return rate;
+    }
+
+    public Arrival getArrival() {
+        return arrival;
+    }
+
     public List<Double> getWeights() {
         return (this.weights);
     }
-    
+
     /**
      * Computes the sum of weights. Usually needs to add up to 100%
-     * 
+     *
      * @return The total weight
      */
     public double totalWeight() {
         double total = 0.0;
-        for (Double d : weights)
+        for (Double d : weights) {
             total += d;
+        }
         return total;
     }
-    
+
     /**
      * This simply computes the next transaction by randomly selecting one based
      * on the weights of this phase.
-     * 
+     *
      * @return
      */
     public int chooseTransaction() {
-        return chooseTransaction(false, 0);
+        return chooseTransaction(false);
     }
 
-    public int chooseTransaction(boolean isColdQuery, int terminalId) {
-        if (isDisabled())
+    public int chooseTransaction(boolean isColdQuery) {
+        if (isDisabled()) {
             return -1;
+        }
 
         if (isSerial()) {
             int ret;
-            synchronized(this) {
-                int next = this.nextSerials.get(terminalId);
-                ret = next;
+            synchronized (this) {
+                ret = this.nextSerial;
 
                 // Serial runs should not execute queries with non-positive
                 // weights.
-                while (ret < this.num_weights && weights.get(ret - 1).doubleValue() <= 0.0)
-                    ret = ++next;
+                while (ret <= this.weightCount && weights.get(ret - 1) <= 0.0) {
```

Comment:
TODO unrelated to this pr, the bug with the last query in serial mixed is probably around here